### PR TITLE
Add ability to pre-configure a BR IID for DAGID

### PR DESCRIPTION
**Pull request by simonduq**
For more context see https://github.com/wittra/wittra-firmware/pull/2674

### CODE REVIEW

File: os/net/routing/rpl-lite/rpl.c [hunk 1]
```diff
@@ -82,12 +82,23 @@ rpl_get_global_address(void)
     prefix_length = curr_instance.dag.prefix_info.length;
   }
 
+#ifdef RPL_ROOT_DAG_IID
+  uip_ipaddr_t preset_dag_iid;
+  uint8_t iid[] = RPL_ROOT_DAG_IID;
+  memcpy(preset_dag_iid.u8 + 8, iid, 8);
```
Comment (Beshr-wittra) [inline os/net/routing/rpl-lite/rpl.c:88]: Replace 8 with sizeof(RPL_ROOT_DAG_IID)

**Review by Beshr-wittra — COMMENTED** [review]
Minor comment

File: os/net/ipv6/uip-sr.c [hunk 2]
```diff
@@ -80,6 +83,12 @@ node_matches_address(void *graph, const uip_sr_node_t *node, const uip_ipaddr_t
 }
 /*---------------------------------------------------------------------------*/
 uip_sr_node_t *
+uip_sr_get_root_node(void)
+{
+  return root_node;
```
Comment (Beshr-wittra) [inline os/net/ipv6/uip-sr.c:88]: Would this work?
```suggestion
  static uip_sr_node_t *local_root_node = NULL;
  /* if(local_root_node == NULL) */
  {
    uip_ipaddr_t root_ipaddr;
    uip_sr_node_t *node;
    NETSTACK_ROUTING.get_root_ipaddr(&root_ipaddr);
    node = uip_sr_get_node(graph, addr);
    local_root_node = uip_sr_get_node(graph, &root_ipaddr);
  }

  return local_root_node;
```

**Review by Beshr-wittra — COMMENTED** [review]
I am not completely following up here (9af34e18252536ee3a0e4ce9209a2a9c462bc0a9), why do we want to recursively get the root?

File: os/net/routing/rpl-lite/rpl-ext-header.c [hunk 3]
```diff
@@ -81,7 +81,7 @@ rpl_ext_header_srh_get_next_hop(uip_ipaddr_t *ipaddr)
   }
 
 #if RPL_WITH_ROOT
-  root_node = uip_sr_get_node(NULL, &curr_instance.dag.dag_id);
```
Comment (Beshr-wittra) [inline os/net/routing/rpl-lite/rpl-ext-header.c:84]: Can you explain a little bit here? Why do we have to change it in the first place?

**Comment by simonduq** [comment]
I have added a comment that explains the consequences of this PR on 6lowpan address compression.

**Comment by simonduq** [comment]
Closing as this PR was only up for review purposes; we merge it in the wittra-firmware submodule as usual

**Pull request closed without merging**